semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/SemanticdbGen.scala

```diff
@@ -0,0 +1,58 @@
+package scala.meta.internal.semanticdb.javac
+
+import java.nio.file.{Files, Path}
+import javax.lang.model.element.TypeElement
+import scala.collection.mutable
+import scala.meta.internal.index.SemanticdbIndex
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.semanticdb.javac.semantics._
+
+class SemanticdbGen(relSourcePath: Path, toplevels: Seq[TypeElement], index: SemanticdbIndex) {
+
+  private val infos = mutable.ListBuffer[s.SymbolInformation]()
+
+  private lazy val relOutputPath = relSourcePath.toString + ".semanticdb"
+
+  def populate(): Unit = {
+
+    def infoForPackage(pkgSym: String): s.SymbolInformation =
+      s.SymbolInformation(
+        symbol = pkgSym,
+        language = s.Language.JAVA,
+        kind = s.SymbolInformation.Kind.PACKAGE,
+        name = pkgSym.desc.name,
+        accessibility = Some(
+          s.Accessibility(
+            tag = s.Accessibility.Tag.PUBLIC
+          ))
+      )
+
+    val pkg = toplevels.head.enclosingPackage
+
+    toplevels.foreach { toplevel =>
+      val toplevelInfo = toplevel.populateInfos(infos)
+      index.append(relOutputPath, List(toplevelInfo))
+    }
+
+    pkg.sym.ownerChain.foreach { packageName =>
+      infos += infoForPackage(packageName)
+    }
+  }
+
+  def persist(targetRoot: Path): Unit = {
+    val outputFile = targetRoot.resolve("META-INF/semanticdb").resolve(relOutputPath)
+    Files.createDirectories(outputFile.getParent)
+    val outputDocuments = s.TextDocuments(
+      List(
+        s.TextDocument(
+          schema = s.Schema.SEMANTICDB4,
+          uri = relSourcePath.toString,
```

Comment:
Missing `PathIO.toUnix(...)` to work on Windows. Note to self that `RelativePath.toURI` would indeed be nice (https://github.com/scalameta/scalameta/issues/1523)!